ui-tests/src/test/java/org/openmrs/reference/page/PatientVisitsDashboardPage.java

```diff
@@ -20,8 +20,7 @@
 	
     private static final By CAPTURE_VITALS = By.id("referenceapplication.realTime.vitals");
     private static final By VISIT_LIST = By.cssSelector("#visits-list li.menu-item.viewVisitDetails span.menu-date");
-    private static final By END_VISIT = By.cssSelector("#visit-details div.visit-actions.active-visit a:nth-child(1)");
-    private static final By END_VISIT_DIALOG = By.id("end-visit-dialog");
+    private static final By END_VISIT = By.className("icon-off");
```

Comment:
After trying most element attribute property values (`id, css, xpath, linkText, className, partialLinkText`)  in retrieving the element I found two best alternatives;` By.className()` and `By.linkText();`. Technically className overrides LinkText() for code sustainability